iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -1231,6 +1238,25 @@ private void HandleConnectionStatusExceptions(Exception exception, bool retryAtt
                     nameof(HandleConnectionStatusExceptions));
         }
 
+        private bool IsClientTransportOpen()
+        {
+            return Interlocked.Read(ref _isOpen) == 1;
+        }
+
+        private void SetClientTransportStatus(ClientTransportStatus transportStatus)
+        {
+            switch (transportStatus)
```

Comment:
Any reason why you are using a switch here? If/Else is usually what I go for unless there are a lot of possible cases or if I need fall through.

Reply:
I agree. I left this as a switch case since I think I might need to add another case here to facilitate the open-close-open flow that we have in previews/v2. I left the switch here as well, just for keeping the code the same. That said, if I don't end up using 2+ cases on previews/v2 I'll update both to use simpler if-else instead.